Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h

```diff
@@ -246,6 +246,9 @@ WK_CLASS_AVAILABLE(macos(13.3), ios(16.4))
  */
 @property (nonatomic, readonly) BOOL hasCommands;
 
+/*! @abstract A boolean value indicating whether the extension includes declarativeNetRequest rules used for content blocking. */
+@property (nonatomic, readonly) BOOL hasDeclarativeNetRequestRules;
```

Comment:
I would like to avoid `declarativeNetRequest` in our API. The extension API name is a bit on the nose, and our API should be more generic to be able to morph and adapt as the extension ecosystem evolves.

Maybe:

```
/*! @abstract A boolean value indicating whether the extension includes rules used for content modification or blocking. */
@property (nonatomic, readonly) BOOL hasContentModificationRules;
```

This would allow it to encompass any declarative style rules, since that might be under a new name based on talks in the WECG.

Reply:
Sounds good!